x-pack/plugins/infra/public/containers/metrics_explorer/with_metrics_explorer_options_url_state.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+import { set, values } from 'lodash';
+import React, { useContext, useMemo } from 'react';
+import { MetricsExplorerColor } from '../../../common/color_palette';
+import { MetricsExplorerAggregation } from '../../../server/routes/metrics_explorer/types';
+import { UrlStateContainer } from '../../utils/url_state';
+import {
+  MetricsExplorerOptions,
+  MetricsExplorerOptionsContainer,
+  MetricsExplorerTimeOptions,
+} from './use_metrics_explorer_options';
+
+interface MetricsExplorerUrlState {
+  timerange?: MetricsExplorerTimeOptions;
+  options?: MetricsExplorerOptions;
+}
+
+export const WithMetricsExplorerOptionsUrlState = () => {
+  const { options, currentTimerange, setOptions: setRawOptions, setTimeRange } = useContext(
+    MetricsExplorerOptionsContainer.Context
+  );
+
+  const setOptions = (value: MetricsExplorerOptions) => {
+    setRawOptions(value);
+  };
+
+  const urlState = useMemo(
+    () => ({
+      options,
+      timerange: currentTimerange,
+    }),
+    [{ options, currentTimerange }]
```

Comment:
Not sure if I'm misunderstanding something here, but I'd have expected the dependency array to be an array of separate values, rather than combined in one object. E.g. `[options, currentTimerange]`, so if one of those distinct values changes the memoized value is recalculated.

Reply:
Nope... brain was on autopilot.